flow/project.py

```diff
@@ -1603,19 +1470,25 @@ def _add_dummy_operation(job):
         op_counter = Counter()
         for job in context['jobs']:
             for k, v in job['operations'].items():
-                if v['eligible']:
+                if k != '' and v['eligible']:
                     op_counter[k] += 1
         context['op_counter'] = op_counter.most_common(eligible_jobs_max_lines)
         n = len(op_counter) - len(context['op_counter'])
         if n > 0:
             context['op_counter'].append(('[{} more operations omitted]'.format(n), ''))
 
-        print(template.render(**context), file=file)
+        render_result = render_status.Renderer()
+        render_output = render_result.render(template, template_environment, context, detailed,
+                                             expand, unroll, compact, output_format)
+
+        print(render_output, file=file)
 
         # Show profiling results (if enabled)
         if profiling_results:
             print('\n' + '\n'.join(profiling_results), file=file)
 
+        return render_result
```

Comment:
Why do we return that?

Reply:
I think this was from a discussion with @vyars awhile ago. The idea was that now that we have separate the format rendering logic into the renderer class, we can provide this directly to user to allow them more flexibility to render different format without repeatedly calling print_status.

Reply:
Ok, fair. I can see some use case where one would take advantage of that in the context of dashboard for example. I'm not 100% sure that this it the right level of abstraction, but I also don't really see a strong reason against this.